schemes/radiation_utils/solar_irradiance_data.F90

```diff
@@ -0,0 +1,404 @@
+    class(abstract_netcdf_reader_t), pointer :: file_reader
+    integer, parameter :: missing_variable_error_code = 3
+
+    character(len=*), parameter :: subname = 'solar_irradiance_data_register: '
+
+    ! Set error variables
+    errmsg = ''
+    errflg = 0
+
+    nbins = 0
+    nbinsp = 0
+
+    ! Check if irradiance file path is set.
+    ! If not then assume that a constant
+    ! solar flux will be set:
+    if (trim(irrad_file_path) == 'NONE') then
+       fixed_scon = .true.
+       return !Nothing else to do in the register phase.
+    end if
+
+    file_reader => create_netcdf_reader_t()
+
+    ! Open the solar irradiance data file
+    call file_reader%open_file(irrad_file_path, errmsg, errflg)
+    if (errflg /= 0) then
+       errmsg = subname // errmsg
+       return
+    end if
+
+    ! Read the wavelengths variable
+    call file_reader%get_var('wavelength', lambda, errmsg, errflg)
+    if (errflg /= 0 .and. errflg /= missing_variable_error_code) then
+       errmsg = subname // errmsg
+       return
+    else if (errflg == missing_variable_error_code) then
+       ! Check old name (for backward compatibility
+       call file_reader%get_var('wvl', lambda, errmsg, errflg)
+    end if
+
+    ! Close the solar irradiance file
+    call file_reader%close_file(errmsg, errflg)
+    if (errflg /= 0) then
+       errmsg = subname // errmsg
+       return
+    end if
+    deallocate(file_reader)
+    nullify(file_reader)
+
+    if (errflg /= 0) then
+       ! Override the errflg, it's ok if there is no wavelength info on file in some scenarios
+       errflg = 0
+       return
+    end if
+
+    ! Set output variables (dimensions)
+    nbins = size(lambda)
+    nbinsp = nbins + 1
+
+  end subroutine solar_irradiance_data_register
+!> \section arg_table_solar_irradiance_data_init Argument Table
+!! \htmlinclude solar_irradiance_data_init.html
+!!
+  subroutine solar_irradiance_data_init(irrad_file_path, solar_data_type, solar_data_ymd, solar_data_tod, solar_const, &
+                  solar_heating_spectral_scl, speed_of_light, planck_const, nbins, nbinsp, do_spectral_scaling, has_spectrum, sol_tsi, &
+                  we, sol_irrad, errmsg, errflg)
+    use ccpp_io_reader,   only: abstract_netcdf_reader_t, create_netcdf_reader_t
+    ! Arguments
+    character(len=*), intent(in) :: irrad_file_path
+    character(len=*), intent(in) :: solar_data_type
+    integer, intent(in) :: solar_data_ymd
+    integer, intent(in) :: solar_data_tod
+    real(kind_phys), intent(in) :: solar_const
+    logical, intent(in) :: solar_heating_spectral_scl
+    real(kind_phys), intent(in) :: speed_of_light
+    real(kind_phys), intent(in) :: planck_const
+    logical, intent(out) :: do_spectral_scaling   ! flag to do spectral scaling
+    logical, intent(out) :: has_spectrum        ! flag for whether solar input file has irradiance spectrum
+    real(kind_phys), intent(out) :: sol_tsi
+    real(kind_phys), allocatable, intent(out) :: we(:)
```

Comment:
Can we give this variable a more meaningful name, or possibly even use the full standard name instead?

```suggestion
    real(kind_phys), allocatable, intent(out) :: wavelength_endpoints(:)
```

Reply:
good call - changed to "wavelength_endpoints"